http/src/main/java/io/micronaut/http/cookie/CookieHttpCookieAdapter.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.cookie;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import java.net.HttpCookie;
+import java.util.Optional;
+
+/**
+ * Adapt from {@link HttpCookie} to {@link Cookie}.
+ * @author Sergio del Amo
+ * @since 4.3.0
+ */
+@Internal
+public class CookieHttpCookieAdapter implements Cookie {
```

Comment:
can this be `final`? Does it need to be `public`?

Reply:
> can this be final? 

https://github.com/micronaut-projects/micronaut-core/pull/10402/commits/a8d3204d9c512fcc50950358fd14bf25905b6bbe

> Does it need to be public?

[I removed the public modifier](https://github.com/micronaut-projects/micronaut-core/pull/10402/commits/f4a6f90d37608c74172a85f2d9583f20f09cc925)